derp/derp.go

```diff
@@ -101,6 +101,19 @@ const (
 
 	framePing = frameType(0x12) // 8 byte ping payload, to be echoed back in framePong
 	framePong = frameType(0x13) // 8 byte payload, the contents of the ping being replied to
+
+	// frameHealth is sent from server to client to tell the client
+	// if they connection is unhealthy somehow. Currently the only unhealthy state
+	// is whether the connection is detected as a duplicate.
+	// The entire frame body is the text of the error message. An empty message
+	// clears the error state.
+	frameHealth = frameType(0x14)
+
+	// frameRestarting is sent from server to client for the
+	// server to declare that it's restarting. Payload is two big
+	// endian uint32 duration in milliseconds: when to reconnect,
+	// and how long to try total.
```

Comment:
What's the purpose of "how long to try total"? If that time period expires, what's the alternative? And if there is a valid alternative, is the client allowed to try that alternative before this period as ended?

Reply:
It's explained more on the ServerRestartingMessage field docs. I'll reference those.